components/shared/ui/menu.tsx

```diff
@@ -1,38 +1,105 @@
+  )
+}
+
 export const Menu: FC<Props> = props => {
   const siteCodename = useSiteCodename();
+  const [activeMenu, setActiveMenu] = useState<string | number>();
+  const [smallMenuActive, setSmallMenuActive] = useState(false);
+
+  const handleMenuClick = (menuId: string | number) => {
+    menuId === activeMenu ? setActiveMenu(-1) : setActiveMenu(menuId);
+  }
 
   return (
-    <div className={`flex flex-auto h-16 w-full ${mainColorBgClass[siteCodename]} justify-between shadow-md items-center px-10 grow-0 shrink-0`}>
-      <div />
-      <menu className="h-full grow-0 py-1 flex flex-auto shrink-0 justify-end items-center gap-x-5">
-        {props.links.map(link => (
-          <li
-            key={link.title}
-            className="h-full"
-          >
-            <Link
-              className="h-full px-5 flex items-center cursor-pointer rounded hover:bg-white"
-              href={link.url}
-            >
-              {link.title}
+    <div className={`w-screen ${mainColorBgClass[siteCodename]} fixed z-50`} {...createItemSmartLink(props.item.system.id)}>
+      <div className="flex justify-between items-center mx-auto max-w-screen-xl md:h-16">
+        <div className="w-screen h-full md:flex justify-between z-50">
+          <div className="flex h-full justify-between items-center px-8 py-4">
+            <Link href="/" className="flex items-center">
+              <span className="font-extrabold">HealthTech</span>
             </Link>
-          </li>
-        ))}
-      </menu>
+            <button
+              type="button"
+              className="md:hidden flex justify-center items-center"
+              onClick={() => setSmallMenuActive(!smallMenuActive)}
+            >
+              <Bars3Icon className="w-6 h-6" />
+            </button>
+          </div>
+          <div>
+            <ul className={`${smallMenuActive ? "flex" : "hidden"} md:flex flex-col font-medium md:flex-row h-full`}>
+              {props.item.elements.subitems.linkedItems.map((link, i) => (
+                <li key={i} className="h-full px-5 bg-green-300 group md:hover:bg-white" onClick={() => handleMenuClick(i)}>
+                  {link.elements.subitems.value.length > 0 ? (
+                    <span>
+                      <DropdownButton item={link} />
+                      <div
+                        className={`${i === activeMenu ? "block" : "hidden"} md:group-hover:block absolute z-50 left-0 shadow-sm bg-white border-gray-200 w-full`}
+                      >
+                        <div className="grid max-w-screen-xl px-4 py-5 mx-auto text-gray-900 sm:grid-cols-2 md:grid-cols-3 md:px-6">
+                          <DropdownMenu links={link.elements.subitems.linkedItems} />
+                        </div>
+                      </div>
+                    </span>
+                  ) : (
+                    <Link
+                      {...link.elements.openInANewWindow.value ? { rel: "noopener noreferrer", target: "_blank" } : {}}
```

Comment:
It seems you missed this place with the multi-option element value access.

Reply:
fixed